source/dofs/dof_accessor_set.cc

```diff
@@ -41,91 +41,249 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+template <typename Number>
+DeclException2(ExcNonMatchingElementsSetDofValuesByInterpolation,
+               Number,
+               Number,
+               << "Called set_dof_values_by_interpolation(), but"
+               << " the element to be set, value " << std::setprecision(16)
+               << arg1 << ", does not match with the non-zero value "
+               << std::setprecision(16) << arg2 << " already set before.");
+
+namespace internal
+{
+  /**
+   * In the set_dof_values() we need to invoke abs() also on unsigned data
+   * types, which is ill-formed on newer C++ standards. To avoid this, we use
+   * std::abs on default types but simply return the number on unsigned types
```

Comment:
```suggestion
   * In set_dof_values(), we need to invoke abs() also on unsigned data
   * types, which is ill-formed on newer C++ standards. To avoid this, we use
   * std::abs on default types, but simply return the number on unsigned types.
```